ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/EventsTable.scala

```diff
@@ -72,6 +81,52 @@ private[events] trait EventsTable {
       }
     }
 
+    def toTransactionTree(events: List[Entry[TreeEvent]]): Option[GetTransactionResponse] =
+      events.headOption.map(
+        first => {
+          val (eventsById, rootEventIds) = treeOf(events)
+          GetTransactionResponse(
+            transaction = Some(
+              ApiTransactionTree(
+                transactionId = first.transactionId,
+                commandId = first.commandId,
+                workflowId = first.workflowId,
+                effectiveAt =
+                  Some(Timestamp.of(seconds = first.ledgerEffectiveTime.getTime, nanos = 0)),
```

Comment:
`Date#getTime` returns number of milliseconds since epoch. I would suggest reading `Entry#ledgerEffectiveTime` as `java.time.Instant` and then use `Instant#getEpochSecond` and `Instant#getNano` to construct the protobuf `Timestamp`. Same goes for the `effectiveAt` in `toFlatTransaction`.
```suggestion
                  Some(Timestamp.of(seconds = first.ledgerEffectiveTime.getEpochSecond, nanos = first.ledgerEffectiveTime.getNano)),
```

Reply:
Very good catch, it's missing from the tests, adding this.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5089/commits/f4d624df866815c079a899a2c45497bb6b7ee6e5